### DISCUSSION TITLE
Generic h5 interface that fits different vendors

### DISCUSSION THREAD

**Discussion by ralfHielscher** [mtex-toolbox/mtex#2098] [Feature Requests]
The goal is to rewrite the MTEX interface to load EBSD data from h5 files in a more generic way, such that it

- works with as many vendors as possible
- is robust to small changes across different versions from the same vendor 
- is easy to adapt to new / changed formats
- allows to select additional information to be imported on demand

This is the current list of vendors we aim to support:

- Oxford (.h5oina)
- Brucker
- Bluequartz (.dream3d)
- EDAX

Please comment below if we are missing an h5 file format or if you can provide some sample data.

Thank you very much.

Ralf.

**Comment by kilir** [comment]
Good idea.
Here's the documentation for h5oina:
https://github.com/oinanoanalysis/h5oina/blob/master/H5OINAFile.md
loadEBSD_h5oina currently supports EBSD/EDS, processed EBSD datasets, EM images, except EBSPs. 

There's also this format with a rudimentary loader in this thread which seems to be an older version of edax h5 which seems to be different from more recent(?) edax hdf5 formats.
https://github.com/mtex-toolbox/mtex/discussions/1994#discussioncomment-7663871

Cheers,
Rüdiger

**Comment by Tijmenvermeij** [comment]
Great initiative Ralf!

I think it was already briefly discussed, but it would be nice if the .h5 files generated by EMSphInx could also be included. Lukas Richter made an import script for that, which he gave to me. I think he was also mentioning that he wanted to add it to Mtex?

Best,
Tijmen
